Las/Client.cs

```diff
@@ -0,0 +1,254 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Text;
+
+using RestSharp;
+using Newtonsoft.Json;
+
+using Lucidtech.Las.Utils;
+using Lucidtech.Las.Core;
+
+namespace Lucidtech.Las
+{
+    /// <summary>
+    /// A low level client to invoke api methods from Lucidtech AI Services.
+    /// </summary>
+    ///            
+    public class Client 
+    {
+        /// Domain endpoint of the api, e.g. https://&lt;prefix&gt;.api.lucidtech.ai/&lt;version&gt;
+        private string Endpoint { get; }
+        ///The version of the service, e.q. "v1"
+
+        /// The base client that is used to access the REST API
+        private RestClient RestSharpClient { get; }
+
+        /// The serializer that is used to convert between json format and standard C# types like object and Dictionary
+        public JsonSerialPublisher Serializer{ get; }
+        /// The component that deals with the authorization against AWS 
+        private AmazonAuthorization Authorization { get; }
+
+        public Client(string endpoint, Credentials credentials)
+        {
+            Authorization = new AmazonAuthorization(credentials);
+            Endpoint = endpoint;
+            var uri = new Uri(endpoint);	
+            RestSharpClient = new RestClient(uri.GetLeftPart(UriPartial.Authority));
+            Serializer = new JsonSerialPublisher(new JsonSerializer());
+        }
+        
+        public Client(string endpoint) : this(endpoint, new Credentials()) { }
+
+        /// <summary>
+        ///	Creates a document handle, calls the POST /documents endpoint
+        /// </summary>
+        /// <example>
+        /// Create a document handle for a jpeg image
+        /// <code>
+        /// Client client = new Client();
+        /// var response = client.PostDocuments("image/jpeg", "bar");
+        /// var dictResponse = Client.ObjectToDict&lt;Dictionary&lt;string, string&gt;&gt;(response);
+        /// </code>
+        /// </example>
+        /// <param name="contentType"> A mime type for the document handle </param>
+        /// <param name="consentId"> An identifier to mark the owner of the document handle</param>
+        ///            
+        /// <returns>
+        /// A deserialized object that can be interpreted as a Dictionary with the fields
+        /// with documentId, uploadUrl, contentType and consentId
+        /// </returns>
+        ///         
+        public object PostDocuments(string contentType, string consentId)
+        {
+            var dictBody = new Dictionary<string, string>() { {"contentType", contentType}, {"consentId", consentId} };
+            
```

Comment:
Since each line represents separate steps I have done it this way. It may look strange since it is whitespaces, but I think it make sense when you look at ClientRestRequest. 

It would be nice to discuss this.